### PR TITLE
DHFPROD-1816: Mapping configuration

### PR DESCRIPTION
**Pull request by wooldridge**
Finish mapping config UI work for 5.0, includes:

- Auto-save of mapping changes
- Update mapping config view when mapping settings change
- Support defining source for mapping via cts query
- Validate mapping step data upon creation (and any type pf step data for that matter)
- algorithmRef bug fix
- Fix radio button color

### CODE REVIEW

File: web/src/main/ui/app/components/flows-new/edit-flow/mapping/ui/mapping-ui.component.html [hunk 1]
```diff
@@ -34,7 +28,7 @@
       </div>
 
       <!-- Dropdown row for each entity property -->
-      <div *ngFor="let entityProp of chosenEntity?.definition.properties; let i = index;" class="src-prop-container">
```
Comment (clockworked247) [inline web/src/main/ui/app/components/flows-new/edit-flow/mapping/ui/mapping-ui.component.html]: Let's leave these selectors the same.   Wiki and QA already using the old one.  Less churn.

Comment (wooldridge) [reply web/src/main/ui/app/components/flows-new/edit-flow/mapping/ui/mapping-ui.component.html]: Can you tell me exactly what to change back? I'm not sure what I did. :-)

Comment (ayuwono) [reply web/src/main/ui/app/components/flows-new/edit-flow/mapping/ui/mapping-ui.component.html]: @wooldridge , it's the class name, the old one is src-prop-container. @clockworked247 , I think it's okay to have the new name source-prop-container so it's consistent and more readable? Wiki and QA can change the naming

Comment (wooldridge) [reply web/src/main/ui/app/components/flows-new/edit-flow/mapping/ui/mapping-ui.component.html]: @clockworked247 sounds like @ayuwono doesn't need this changed?

Comment (clockworked247) [reply web/src/main/ui/app/components/flows-new/edit-flow/mapping/ui/mapping-ui.component.html]: Ok, just update then wiki for me then.

**Comment by wooldridge** [comment]
@alexsmr, THANK YOU.

**Pull request merged**